benchmarks/dynamo/common.py

```diff
@@ -1222,6 +1222,31 @@ def __init__(
             self.model_dir / f"{model_name}_{self._COMPILER_NAME}.onnx"
         )
 
+    def _determine_deepcopy_target_device(self):
+        if current_device == "cpu":
+            target_device = "cpu"
+        else:
+            if torch.cuda.device_count() > 1:
+                # Copy to another cuda device to avoid OOM.
+                target_device = "cuda:1"
```

Comment:
what if the user does CUDA_VISIBLE_DEVICE="2,3"?

By hardcoding a random device ID, things can go south. Maybe try to read the env var to see which devices are available. or use torch.cuda to get this list

Reply:
it doesn't work that way. For `export CUDA_VISIBLE_DEVICE="2,3"` case, the available device id is still "0, 1" from code perspective.